### PR TITLE
[OsLoader] Clean-up PreOsChecker

### PR DESCRIPTION
**Pull request by myguitar**
Split PreOsChecker in a separate file and add minimum APIs
- IsPreOsCheckerLoaded ()
- LoadPreOsChecker ()
- StartPreOsChecker ()

Signed-off-by: Aiden Park <aiden.park@intel.com>

### CODE REVIEW

File: PayloadPkg/OsLoader/PreOsChecker.c [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+  PreOsParams.OsBootState.Esi     = (UINT32) OsBootParam;
+  PreOsParams.OsBootState.Eip     = OsBootParam->Hdr.Code32Start;
+  PreOsParams.OsBootState.Eflags  = 0;
+
+  EntryPoint = (PRE_OS_CHECKER_ENTRY)mPreOsCheckerEntry;
+  EntryPoint (&PreOsParams);
+
+  //
+  // Not reach here
```
Comment (jgutbub) [inline PayloadPkg/OsLoader/PreOsChecker.c]: Should we add a CPU dead loop or ASSERT here for debug?

Comment (myguitar) [reply PayloadPkg/OsLoader/PreOsChecker.c]: Updated in coming push. Thanks.

File: PayloadPkg/OsLoader/PreOsChecker.c [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  //  DEBUG ((DEBUG_INFO, "Pre-OS checker is already loaded at entry @ 0x%p\n", mPreOsCheckerEntry));
+  //  return EFI_STATUS;
+  //}
+
+  //
+  // Check if there is a pre-OS checker that needs to be executed
+  //
+  Buffer = NULL;
+  Length = 0;
+  Status = LoadComponent (SIGNATURE_IPFW, SIGNATURE_POSC, &Buffer, &Length);
```
Comment (mauricema) [inline PayloadPkg/OsLoader/PreOsChecker.c]: Do we have a way to indicate if PreOsChecker support is required or not ?   
If a platform does not need PreOsChecker,  maybe we should not try to load the image at all.

Comment (myguitar) [reply PayloadPkg/OsLoader/PreOsChecker.c]: Updated in coming push. Token space has been updated. Will follow up on-going boards which will be impacted by this change once this is merged.

**Pull request merged**